### PR TITLE
VOCKQJK11LM Improvements and modernExtend conversion (2/4)

### PR DESCRIPTION
**Pull request by sjorge**
Still very WIP but I though I'd open as draft already.

I plan on extending the modernExtend numeric with a 'calibration' argument that will toggle the calibration options.
It would be nice to have that code available for all numerics so we can easily enable where it makes sense.

### CODE REVIEW

File: src/lib/modernExtend.ts [hunk 1]
```diff
@@ -483,3 +483,34 @@ export function forceDeviceType(args: {type: 'EndDevice' | 'Router'}): ModernExt
...
     return {configure, isModernExtend: true};
 }
+
+// XXX: add options
+export function temperature(args?: Partial<NumericArgs>) {
+    return numeric({
+        name: 'temperature',
+        cluster: 'msTemperatureMeasurement',
+        attribute: 'measuredValue',
+        configureReporting: {minimumReportInterval: 10, maximumReportInterval: repInterval.HOUR, reportableChange: 100},
```
Comment (sjorge) [inline src/lib/modernExtend.ts:493]: I took the same values as in src/lib/reporting.ts

File: src/devices/xiaomi.ts [hunk 2]
```diff
@@ -3109,20 +3112,20 @@ const definitions: Definition[] = [
...
         description: 'Aqara TVOC air quality monitor',
-        fromZigbee: [fz.battery, fz.temperature, fz.humidity, xiaomi.fromZigbee.aqara_opple],
+        fromZigbee: [fz.battery, xiaomi.fromZigbee.aqara_opple],
         meta: {battery: {voltageToPercentage: '3V_2850_3000'}},
-        exposes: [e.temperature(), e.humidity(), e.device_temperature(), e.battery(), e.battery_voltage()],
+        exposes: [e.device_temperature(), e.battery(), e.battery_voltage()],
         extend: [
             aqaraAirQuality(),
             aqaraVoc(),
+            temperature({endpointID: 1}),
```
Comment (Koenkk) [inline src/devices/xiaomi.ts:3121]: What is the data/database.db entry when pairing this device?

File: src/lib/modernExtend.ts (hunk 1 above)
Comment (Koenkk) [reply src/lib/modernExtend.ts:493]: I think you also need ` scale: 100,` here?

Comment (sjorge) [reply src/lib/modernExtend.ts:493]: Yes, It's already fixed locally. I'll rebase and push, still working on the moving over the calibration code.

**Comment by sjorge** [comment]
@Koenkk looks like somehow with the current code the options for temperature calibration are already somehow exposed. I don't understand where it's getting it from though.

**Comment by Koenkk** [comment]
@sjorge I don't understand it either, but anyway the whole calibration should be moved up (outside of the converters)

**Comment by sjorge** [comment]
> @sjorge I don't understand it either, but anyway the whole calibration should be moved up (outside of the converters)

Where do you want the calibration to end up ?

**Comment by sjorge** [comment]
> There should be some "post process" function (out of scope for this PR), thinking about it now, you can remove the whole calibrate from your code, since the calibrate options will be discovered automatically based on the exposes.

Having a post process chain would be very nice, it could also fix things like `fz.SNZB_temperature()` as we can just add filter function to the chain.

So I just drop the calibrate stuff and the PR is good to go I guess ?

**Comment by sjorge** [comment]
Cleaned up the calibrate stuff.

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**